R/integrations.R

```diff
@@ -161,3 +161,115 @@ basicSeuratProc <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, do
     umap = umap
   )
 }
+
+#' RNA velocity analysis on samples integrated with conos
+#'
+#' @description Create a list of objects to pass into gene.relative.velocity.estimates function from the velocyto.R package
+#'
+#' @param cms_list list of velocity files written out as cell.counts.matrices.rds files by running dropest with -V option
+#' @param con conos object (after creating an embedding and running leiden clustering)
+#' @param n.odgenes number of overdispersed genes to use for PCA
+#' @param verbose verbose mode
+#'
+#'
+#' @return List with cell distances, combined spliced expression matrix, combined unspliced expression matrix, combined matrix of spanning reads, cell colors for clusters and embedding (taken from conos)
+#'
+#'
+#'
+#' @export
+#'
+VelocityInfoConos <- function(cms_list, con, n.odgenes = 2e3, verbose = TRUE, min.max.cluster.average.emat=0.2, min.max.cluster.average.nmat=0.05, min.max.cluster.average.smat=0.01) {
+  if (!requireNamespace("velocyto.R")) {
+    stop("You need to install 'velocyto.R' package to be able to use this function")
+  }
+
+  if (verbose) cat("Merging raw count matrices...\n")
+  # Merge samples to get names of relevant cells and genes 
+  raw.count.matrix.merged <- lapply(con$samples, conos:::getRawCountMatrix) %>%
+    conos:::mergeCountMatrices()
+  
+  if (verbose) cat("Merging velocity files...\n")
+  # Intersect genes and cells between the conos object and all the velocity files
+  cms_list <- lapply(cms_list, PrepareVelocity, rownames(raw.count.matrix.merged), colnames(raw.count.matrix.merged))
+  # Keep only genes present in velocity files from all the samples
+  common_genes <-  Reduce(intersect, lapply(cms_list, function(x) {rownames(x[[1]])}))
+  cms_list = lapply(cms_list, function(x) {lapply(x, function(y) {y[row.names(y) %in% common_genes,]} )} )
+  
+  # Merge velocity files from different samples
+  emat <- do.call(cbind, lapply(cms_list, function(x) {x[[1]]}))
+  nmat <- do.call(cbind, lapply(cms_list, function(x) {x[[2]]}))
+  smat <- do.call(cbind, lapply(cms_list, function(x) {x[[3]]}))
+  
+  cluster.label <- con$clusters$leiden$groups
+  cell.colors <- pagoda2:::fac2col(cluster.label)
```

Comment:
We [have](https://github.com/hms-dbmi/conos/blob/95c33d7120b7f330195cc7ee4927881976517cbf/R/plot.R#L492) this function in conos, and pagoda2 is optional dependency. Please, remove `pagoda2:::`.